certora/specs/SocialRecoveryModule.spec

```
@@ -1,26 +1,208 @@
+
+    assert safeContract.isModuleEnabled(currentContract);
+    assert e.msg.sender == safeContract;
+    assert currentContract.isGuardian(safeContract, guardian);
+    assert guardian != otherAccount => otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount + 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies that the guardian can always be added considering ideal conditions.
+rule guardianCanAlwaysBeAdded(env e, address guardian, uint256 threshold) {
+    requireSocialRecoveryModuleEnabled();
+
+    // No value should be sent with the transaction.
+    require e.msg.value == 0;
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;    
+    // The guardian count should be less than the maximum value to prevent overflow.
+    require currentGuardiansCount < max_uint256; // To prevent overflow (Realistically can't reach).
+    requireGuardiansLinkedListIntegrity(guardian);
+
+    // The guardian should not be values such as zero, sentinel, or the Safe contract itself.
+    require guardian != 0;
+    require guardian != SENTINEL();
+    require guardian != safeContract;
+
+    // The guardian should not be an owner of the Safe contract at the time of addition.
+    require !safeContract.isOwner(guardian);
+    // The guardian should not be already added as a guardian.
+    require !currentContract.isGuardian(safeContract, guardian);
+
+    // The threshold must be greater than 0 and less than or equal to the total number of guardians.
+    require threshold > 0 && to_mathint(threshold) <= guardianStorageContract.entries[safeContract].count + 1;
+
+    // Safe contract should be the sender of the transaction.
+    require e.msg.sender == safeContract;
+    currentContract.addGuardianWithThreshold@withrevert(e, safeContract, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert !isReverted &&
+        currentContract.isGuardian(safeContract, guardian) &&
+        currentGuardiansCount + 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies the possibilites in which the addition of a new guardian can revert.
+rule addGuardianRevertPossibilities(env e, address guardian, uint256 threshold) {
+    bool isGuardian = currentContract.isGuardian(safeContract, guardian);
+
+    currentContract.addGuardianWithThreshold@withrevert(e, safeContract, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert isReverted =>
+        isGuardian ||
+        e.msg.sender != safeContract ||
+        e.msg.value != 0 ||
+        guardian == 0 ||
+        guardian == SENTINEL() ||
+        guardian == safeContract ||
+        safeContract.isOwner(guardian) ||
+        threshold == 0 ||
+        threshold >= guardianStorageContract.entries[safeContract].count ||
```

I think you missed a spot here from the previous comment about the maximum threshold.

```suggestion
        threshold > guardianStorageContract.entries[safeContract].count + 1 ||
```

---

Done 👍🏾